text/0000-cargo-prepublish.md

```diff
@@ -0,0 +1,395 @@
+allowed replacing existing versions in a registry, but issued a warning when
+doing so. It turned out that without a warning we ended up having a full-on
+`[replace]` replacement!
+
+At this time, though, it is not planned to deprecate the `[replace]` section,
+nor remove it. After the `[augment]` section is implemented, if it ends up
+working out this may change. If after a few cycles on stable the `[augment]`
+section seems to be working well we can issue an official deprecation for
+`[replace]`, printing a warning if it's still used.
+
+Documentation, however, will immediately being to recommend `[augment]` over
+`[replace]`.
+
+# How We Teach This
+[how-we-teach-this]: #how-we-teach-this
+
+Augmentation is a feature intended for large-scale projects spanning many repos
+and crates, where you want to make something like an atomic change across the
+repos. As such, it should likely be explained in a dedicated section for
+large-scale Cargo usage, which would also include build system integration and
+other related topics.
+
+The mechanism itself is straightforward enough that a handful of examples (as in
+this RFC) is generally enough to explain it. In the docs, these examples should
+be spelled out in greater detail.
+
+Most notably, however, the [overriding dependenices][over] section of Cargo's
+documentation will be rewritten to primarily mention `[augment]`, but
+`[replace]` will be mentioned still with a recommendation to use `[augment]`
+instead if possible.
+
+[over]: http://doc.crates.io/specifying-dependencies.html#overriding-dependencies
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This feature adds yet another knob around where, exactly, Cargo is getting its
+source and version information. In particular, it's basically deprecating
+`[replace]` if it works out, and it's typically a shame to deprecate major
+stable features.
+
+Fortunately, because these features are intended to be relatively rarely used,
+checked in even more rarely, are only used for very large projects, and cannot
+be published to crates.io, the knobs are largely invisible to the vast majority
+of Cargo users, who are unaffected by them.
+
+# Alternatives
+[alternatives]: #alternatives
+
+The primary alternative for addressing the motivation of this RFC would be to
+loosen the restrictions around `[replace]`, allowing it to arbitrarily change
+the version of the crate being replaced.
+
+As explained in the motivation section, however, such an approach does not fully
+address the desired workflow, for a few reasons:
+
+- It does not make it possible to track which crates in the dependency graph
+  have successfully upgraded to a new major version of the replaced dependency,
+  which could have the effect of masking important *behavioral* breaking changes
+  (that still allow the crates to compile).
+
+- It does not provide an easy-to-understand picture of what the crates will look
+  like after the relevant dependencies are published. In particular, you can't
```

Comment:
Hm, even with `[augment]`, we're still just providing a picture of what the crates *might* look like after the relevant dependencies are published.... the crate author could choose to publish `1.0.0` instead of `0.10.0`, and there might be more changes than just the ones you're submitting upstream.....

Am I just being a bit too literal in my interpretation of "will" here?